src/jit/gentree.cpp

```diff
@@ -11833,6 +11860,41 @@ void Compiler::gtGetArgMsg(
         }
         else
         {
+#ifdef _TARGET_ARM_
+            if (curArgTabEntry->isSplit)
+            {
+                regNumber argReg = curArgTabEntry->regNum;
+                if (listCount == -1)
+                {
+                    if (curArgTabEntry->numRegs == 1)
+                    {
+                        sprintf_s(bufp, bufLength, "arg%d %s out+%02x%c", argNum, compRegVarName(argReg),
+                                  (curArgTabEntry->slotNum) * TARGET_POINTER_SIZE, 0);
+                    }
+                    else
+                    {
+                        regNumber otherRegNum = (regNumber)((unsigned)(argReg) + curArgTabEntry->numRegs - 1);
+                        char      seperator   = (curArgTabEntry->numRegs == 2) ? ',' : '-';
+                        sprintf_s(bufp, bufLength, "arg%d %s%c%s out+%02x%c", argNum, compRegVarName(argReg), seperator,
+                                  compRegVarName(otherRegNum), (curArgTabEntry->slotNum) * TARGET_POINTER_SIZE, 0);
+                    }
+                }
+                else
+                {
+                    regNumber curReg = (regNumber)((unsigned)(argReg) + listCount);
+                    if (((unsigned)(curReg)) <= MAX_ARG_REG_COUNT)
+                    {
```

Comment:
For example, If `a(int a, int b, struct c)` and `sizeof(struct c)` is 8 byte, `struct c` uses `r2` and `r3`. So we need to compare MAX_ARG_REG_COUNT with the last register, not `listcount`.